src/tsickle.ts

```diff
@@ -225,8 +227,33 @@ function addClutzAliases(
     // default is a keyword in typescript, so the name of the export being default means that it's a
     // default export.
     if (e.name === 'default') return false;
+
     // Otherwise check that some declaration is from the local module.
-    return e.declarations.some(d => d.getSourceFile() === origSourceFile);
+    for (const d of e.declarations) {
+      // This check helps us filter out export * rewritten exports. It appears,
```

Comment:
I think a symbol that is just exported from another file does not have a local declaration. Note that you also cannot reference it locally:

```
export {Foo} from './foo';
let f: Foo; // error
```

So it follows their declaration site is the original file.

Reply:
That is not the case, and hence the bug that I am trying to fix. Here is example debug output, printing for each test case the location of the export.declaration[0] (they all have a single declaration).

```
.===IN FILE export.ts
SYMBOL "NUM_CONSTANT" flags:0x2 BlockScopedVariable
decl loc export.ts
SYMBOL "OTHER" flags:0x2 BlockScopedVariable
decl loc export.ts
SYMBOL "NON_DIRECT" flags:0x200000 AliasExcludes
decl loc export.ts
SYMBOL "NEW_NAME" flags:0x200000 AliasExcludes
decl loc export.ts
===IN FILE import_reexport.ts
SYMBOL "NUM_CONSTANT" flags:0x200000 AliasExcludes
decl loc import_reexport.ts
===IN FILE reexport.ts
SYMBOL "NUM_CONSTANT" flags:0x2 BlockScopedVariable
decl loc export.ts
SYMBOL "OTHER" flags:0x2 BlockScopedVariable
decl loc export.ts
SYMBOL "NON_DIRECT" flags:0x200000 AliasExcludes
decl loc export.ts
SYMBOL "NEW_NAME" flags:0x200000 AliasExcludes
decl loc export.ts
===IN FILE reexport_named.ts
SYMBOL "OTHER" flags:0x200000 AliasExcludes
decl loc reexport_named.ts
SYMBOL "NUM_CONSTANT" flags:0x200000 AliasExcludes
decl loc reexport_named.ts
===IN FILE renamed_export.ts
SYMBOL "MOTHER" flags:0x200000 AliasExcludes
decl loc renamed_export.ts
.
```

Only in the case of 'export *' and actual local export the declaration points to the original file. In all other cases the declaration is in the reexporting file, which made me add this extra code using the getExportSpecifierLocalTargetSymbol API.